### PR TITLE
send action metrics to s3

### PR DESCRIPTION
**Pull request by dkonieczek**

### CODE REVIEW

**Comment by dkonieczek** [comment]
Ready for review.
A run using this branch can be seen here: https://github.com/searchspring-implementations/stfrock.com.au/runs/6396473198?check_suite_focus=true

File: action.yml [hunk 1]
```diff
@@ -221,4 +221,9 @@ runs:
           aws cloudfront create-invalidation --distribution-id ${{ inputs.aws-cloudfront-distribution-id }} --paths "/${{ steps.variables.outputs.siteId }}/${{ steps.variables.outputs.branch }}/*"
           if [ -f tests/lighthouse/runs/manifest.json ]; then aws cloudfront create-invalidation --distribution-id ${{inputs.aws-cloudfront-distribution-id }} --paths "/${{ steps.variables.outputs.siteId }}/.lighthouse/${{ steps.variables.outputs.branch }}/manifest.json"; fi;
         fi
-        
\ No newline at end of file
+
+    - run: date -u +"%FT%TZ" | tr -d '\n' > date.txt && npx -y jjo timestamp=@date.txt type=snap-action data=$(npx -y jjo repository=${{ steps.variables.outputs.repository }} conclusion=${{ github.action_status }}) > SnapAction\[${{ steps.variables.outputs.repository }}\]-$(date -u +"%Y_%m_%d_%H%M").json && aws s3 cp --region us-east-1 ./SnapAction*.json s3://datawarehouse-stage/team-metrics-import/
```
Comment (korgon) [inline action.yml:225]: I'm guessing this falling outside of a named step with the `if` is what will cause this to run on both success and failure?

**Review by korgon — COMMENTED** [review]
I'd like to see an example of this logging a failure before approving.

**Comment by korgon** [comment]
Also - and this may be something to add down the road; it would be nice to know which step failed. For example, if the the authentication (siteId verification) failed, that failure step being a piece of the data could be nice to have available without having to dig into the action. Also, perhaps the action run #?

File: action.yml [hunk 2]
```diff
@@ -221,4 +236,29 @@ runs:
...
-        
\ No newline at end of file
+
+    - run: |
+        npx -y jjo \
+          timestamp=$(date -u +"%FT%TZ" | tr -d '\n') \
+          type=snap-action \
+          data=$(npx -y jjo \
+            conclusion=${{ github.action_status }} \
+            durration=$(expr $(date -u +%s) - ${{ steps.variables.outputs.startTime }} ) \
```
Comment (korgon) [inline action.yml:246]: Misspelling of duration?

**Review by korgon — CHANGES_REQUESTED** [review]
This is looking pretty good.

This is a small thing, but if possible I'd like to see our data keys have some standard naming convention. We are using snake_case and camelCase - we should stick with one or the other. My preference would be camelCase for JSON data.

**Review by korgon — APPROVED** [review]
This is going to provide some nice insight for us. Please squash and merge this branch (tons of commits) :)

**Pull request merged**